src/extension/inspire/src/main/java/org/geoserver/inspire/wmts/InspireGridSetLoader.java

```diff
@@ -0,0 +1,55 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.inspire.wmts;
+
+import org.geoserver.gwc.GWC;
+import org.geoserver.platform.ContextLoadedEvent;
+import org.geowebcache.grid.BoundingBox;
+import org.geowebcache.grid.GridSet;
+import org.geowebcache.grid.GridSetFactory;
+import org.geowebcache.grid.SRS;
+import org.springframework.context.ApplicationListener;
+
+/**
+ * Loads the inspire grid set and mark it as non editable by the user.
+ */
+public class InspireGridSetLoader implements ApplicationListener<ContextLoadedEvent> {
```

Comment:
What if we are reloading the data directory, and maybe the data directory is coming from a node that does not have the inspire extensions... how does this play out? Will we lose the gridset definition? Not a big deal, as it means the data dir being reloaded does not contain any cached layer with that gridset to start with I guess.